pennylane_cirq/cirq_device.py

```diff
@@ -69,23 +70,30 @@ class CirqDevice(QubitDevice, abc.ABC):
     short_name = "cirq.base_device"
 
     def __init__(self, wires, shots, analytic, qubits=None):
-        super().__init__(wires, shots, analytic)
-
-        self.circuit = None
 
-        device_wires = self.map_wires(self.wires)
+        if not isinstance(wires, Iterable):
+            # interpret wires as the number of consecutive wires
+            wires = range(wires)
+        num_wires = len(wires)
 
         if qubits:
-            if wires != len(qubits):
+            if num_wires != len(qubits):
                 raise qml.DeviceError(
                     "The number of given qubits and the specified number of wires have to match. Got {} wires and {} qubits.".format(
                         wires, len(qubits)
                     )
                 )
-
-            self.qubits = qubits
         else:
-            self.qubits = [cirq.LineQubit(wire) for wire in device_wires.labels]
+            qubits = [cirq.LineQubit(idx) for idx in range(num_wires)]
+
+        # cirq orders the subsystems based on a total order defined on qubits.
+        # For consistency, this plugin uses that same total order
+        self.qubits = sorted(qubits)
+
+        super().__init__(wires, shots, analytic)
```

Comment:
I had to move down the superclass initialization here, since it is the thing that creates the `wire_map`, but the `wire_map` can't be determined until the cirq qubits have been assigned

Reply:
@mariaschuld might have some ideas on how to improve `Wire`/cirq qubit integration (there was an allusion to it in #39 and #37)

Reply:
I now I understand the following PR cherry picked from this one.